public/app/features/trails/DataTrailsHistory.tsx

```diff
@@ -91,7 +92,19 @@ export class DataTrailHistory extends SceneObjectBase<DataTrailsHistoryState> {
           }
         }
 
+        respondingToTimeChange = true;
+        const previousStep = this.state.currentStep;
         this.addTrailStep(trail, 'time');
+        const newStep = this.state.currentStep;
+
+        // Ensure the previous trail step keeps the previous state
+        this.state.steps[previousStep].trailState.$timeRange?.setState(prevState);
+
+        // Stepping back and forth ensures that the time range will change properly the next time the user manually switches to previousStep
+        this.goBackToStep(previousStep);
+        this.goBackToStep(newStep);
+
```

Comment:
this seems like a very strange way to fix the underlying issue :) 

Any ideas about the underlying cause and why this fixes it?

Reply:
Ok, found a way.

Reply:
Interesting. Although the tests passed, this solution required an additional step to ensure it re-rendered the controls correctly:
- https://github.com/grafana/grafana/pull/86741/commits/fd5773bc028f023868ec1fe930b9f737caec38bb